lib/entities/main_actions.dart

```diff
@@ -47,23 +47,23 @@ class MainActions {
       switch (walletType) {
         case WalletType.bitcoin:
         case WalletType.litecoin:
-          if (DeviceInfo.instance.isMobile) {
-            Navigator.of(context).pushNamed(Routes.onramperPage);
-          } else {
-            final uri = getIt
-                .get<OnRamperBuyProvider>()
-                .requestUrl();
-            await launchUrl(uri);
+          if (viewModel.isEnabledBuyAction) {
```

Comment:
**Minor:**
we can check if the buy / sell are disabled at the beginning of the function and just `return;` if it's disabled instead of nesting if conditions